### PR TITLE
code refactoring for the run button

### PR DESCRIPTION
**Pull request by devoncarew**
- move the logic for updating the run button UI from the editor code to the `playground` class
- have the regular run button (non-error, non-warning) use a color of `#aaa`. The green color looked a bit too muted.

@kasperpeulen

### CODE REVIEW

**Comment by kasperpeulen** [comment]
Maybe use this color green `#5eaf5e`?

image

File: lib/playground.dart [hunk 1]
```diff
@@ -343,9 +343,14 @@ class Playground {
         return new Annotation(issue.kind, issue.message, issue.line,
             start: start, end: end);
       }).toList());
+
+      _updateRunButton(
+          hasErrors: result.issues.any((issue) => issue.kind == 'error'),
+          hasWarnings: result.issues.any((issue) => issue.kind == 'warning'));
```
Comment (kasperpeulen) [inline lib/playground.dart]: cool trick !

Comment (devoncarew) [inline lib/playground.dart]: Yeah, the methods on List and Iterable - `where`, `any`, `map`, `join`, .. - are super useful.

**Comment by devoncarew** [comment]
> Maybe use this color green #5eaf5e?

I'd like to stick to the regular button text color for now; it'll be more consistent with the rest of the UI. Once we're more officially following the material design guidelines, the whole button color will likely be our accent color.

**Pull request merged**